descartes_core/include/descartes_core/path_planner_base.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * path_planner_base.h
+ *
+ *  Created on: Jan 19, 2015
+ *      Author: ros developer 
+ */
+
+#ifndef DESCARTES_CORE_PATH_PLANNER_BASE_H_
+#define DESCARTES_CORE_PATH_PLANNER_BASE_H_
+
+#include <descartes_core/trajectory_pt.h>
+#include <descartes_core/robot_model.h>
+#include <vector>
+
+namespace descartes_core
+{
+
+class PathPlannerBase
+{
+public:
+  virtual ~PathPlannerBase(){}
+
+  /**
+   * @brief Plans a path for the given robot model through the points in the input trajectory.
+   * @param model robot model implementation for which to plan a path
+   * @param sampling This value will be interpreted differently by each planner specialization.
+   */
+  virtual bool initialize(RobotModelConstPtr &model, double sampling = 0.1f) = 0;
+
+  /**
+   * @brief Generates a robot path from the trajectory.
+   * @param traj the points used to plan the robot path
```

Comment:
The way that I envisioned using such error structure in the planPath(...) method would be as follows:
```
int planPath(const std::vector<TrajectoryPt>& traj, PlannerError& error_struct)
```
where the returned int would be an error code defined in an enumeration and the error_struct  would look as follows:
```
struct PlannerError
{
  std::string error_message;
  int error_code;
  TrajectoryPt::ID id
}
```

In any case, the error codes would have to be very generic, the ones that come to mind are unavailable IK, high joint velocities, self collision, collision with external objects, planning time exceeded, incorrect DOFs, mismatching link or tool names.
The error strings could be very helpful when troubleshooting issues with a specific planner implementation.